### PR TITLE
Revisit bridge contract

### LINKED ISSUE #810: Revisit bridge contract and figure out why it recovers wrong addresses
Fix the unknown issue we found and make sure it's working again.

### PR DESCRIPTION
**Pull request by taobun**
fixed #810

### CODE REVIEW

File: chain/client/proof.go [hunk 1]
```diff
@@ -264,12 +264,10 @@ func GetBlockRelayProof(cliCtx context.CLIContext, blockId uint64) (BlockRelayPr
 		msg := vote.SignBytes("bandchain")
 		lr := strings.Split(hex.EncodeToString(msg), hex.EncodeToString(block.Hash()))
 
-		val, ok := leftMsgCount[lr[0]]
-		if ok {
-			leftMsgCount[lr[0]] = val + 1
-		} else {
-			leftMsgCount[lr[0]] = 0
+		if leftMsg != "" && leftMsg != lr[0] {
+			return BlockRelayProof{}, fmt.Errorf("Inconsistency prefix signature byte")
```
Comment (sorawit) [inline chain/client/proof.go]: inconsistent .. bytes

**Review by sorawit — CHANGES_REQUESTED** [review]

File: bridges/evm/test/Bridge.test.js [hunk 2]
```diff
@@ -449,7 +504,11 @@ contract("Bridge", ([_, owner, alice, bob]) => {
         { from: owner }
       );
 
-      (await this.bridge.validatorPowers("0x652D89a66Eb4eA55366c45b1f9ACfc8e2179E1c5"))
+      (
```
Comment (sorawit) [inline bridges/evm/test/Bridge.test.js]: Dont re-format it!

**Review by sorawit — CHANGES_REQUESTED** [review]

File: bridges/evm/test/Bridge.test.js (hunk 2 above)
Comment (taobun) [reply bridges/evm/test/Bridge.test.js]: Fixed

**Review by sorawit — APPROVED** [review]
looks good

**Review by prin-r — APPROVED** [review]

**Pull request merged**